cmd/spiffe-helper/config/config.go

```diff
@@ -140,16 +150,24 @@ func ValidateConfig(c *Config, exitWhenReady bool, log logrus.FieldLogger) error
 		}
 	}
 
-	c.ExitWhenReady = c.ExitWhenReady || exitWhenReady
+	x509Enabled, err := validateX509Config(c)
+	if err != nil {
+		return err
+	}
+
+	jwtBundleEnabled, jwtSVIDsEnabled := validateJWTConfig(c)
 
-	x509EmptyCount := countEmpty(c.SVIDFileName, c.SVIDBundleFileName, c.SVIDKeyFileName)
-	jwtBundleEmptyCount := countEmpty(c.SVIDBundleFileName)
-	if x509EmptyCount == 3 && len(c.JWTSVIDs) == 0 && jwtBundleEmptyCount == 1 {
+	if !x509Enabled && !jwtBundleEnabled && !jwtSVIDsEnabled {
 		return errors.New("at least one of the sets ('svid_file_name', 'svid_key_file_name', 'svid_bundle_file_name'), 'jwt_svids', or 'jwt_bundle_file_name' must be fully specified")
 	}
 
-	if x509EmptyCount != 0 && x509EmptyCount != 3 {
-		return errors.New("all or none of 'svid_file_name', 'svid_key_file_name', 'svid_bundle_file_name' must be specified")
+	if isFlagPassed(daemonModeFlagName) {
```

Comment:
can you unify daemonModeFlag to a single value?
it must be calculated before getting here, so we can just set the value we get

Reply:
Can move to `ParseConfig()`. Is that what you're thinking?

Reply:
sure

Reply:
Moved that logic to a separate function